Applications/LLaMA/jni/main.cpp

```diff
@@ -219,6 +219,39 @@ std::vector<int> generate(float *logits, unsigned int NUM_VOCAB = 0,
   return outputs;
 }
 
+std::vector<int> generate_multi_tokens(
+  float *logits, unsigned int NUM_VOCAB = 0, unsigned int NUM_TARGET_TOKENS = 1,
+  float repetition_penalty = 1, unsigned int *input_ids = nullptr,
+  unsigned int NUM_INPUT_IDS = 0, unsigned int *bad_words_ids = nullptr,
+  unsigned int NUM_BAD_WORDS_IDS = 0) {
+  std::vector<int> outputs;
+  // apply repetition penalty
+  if (repetition_penalty != 1 && input_ids != nullptr && NUM_INPUT_IDS != 0) {
```

Comment:
repetition_penalty is a float value so instead of using != operator please do something like this
`abs(repetition_penalty - 1) < eps`

Reply:
Thanks, I applied it.